### PR TITLE
Added a function `from_existing_collection` in `Qdrant` vector database.

### PR DESCRIPTION
**Pull request by spike-spiegel-21**
Issue: #20514 
The current implementation of `construct_instance` expects a `texts: List[str]` that will call the embedding function. This might not be needed when we already have a client with collection and `path,  you don't want to add any text.

This PR adds a class method that returns a qdrant instance with an existing client.

Here everytime https://github.com/langchain-ai/langchain/blob/cb6e5e56c29477c6da5824c17f1b70af11352685/libs/community/langchain_community/vectorstores/qdrant.py#L1592 `construct_instance` is called, this line sends some text for embedding generation.

### CODE REVIEW

**Comment by spike-spiegel-21** [comment]
@hwchase17 @baskaryan Please check if it is relevant to add this function.

**Review by baskaryan — APPROVED** [review]
cc @kacperlukawski

**Comment by kacperlukawski** [comment]
@Anush008 Could you please have a look?

File: libs/community/langchain_community/vectorstores/qdrant.py [hunk 1]
```diff
@@ -1367,6 +1367,28 @@ def from_texts(
         qdrant.add_texts(texts, metadatas, ids, batch_size)
         return qdrant
 
+    @classmethod
+    def from_existing_index(
```
Comment (Anush008) [inline libs/community/langchain_community/vectorstores/qdrant.py:1371]: As per Qdrant's terminology,
```
from_existing_collection()
```
should be more appropriate.

File: libs/community/langchain_community/vectorstores/qdrant.py [hunk 2]
```diff
@@ -1367,6 +1367,28 @@ def from_texts(
...
+        cls: Type[Qdrant],
+        embedding: Embeddings,
+        path: str,
```
Comment (Anush008) [inline libs/community/langchain_community/vectorstores/qdrant.py:1374]: Why is `path` the only explicit param while the others are `**kwargs`?
I'd recommend specifying all the params supported by `_generate_clients()` explicitly for better typing.

**Review by Anush008 — CHANGES_REQUESTED** [review]
Suggested some QOL improvements.

**Comment by spike-spiegel-21** [comment]
@kacperlukawski @Anush008 thank you for suggesting improvements. I will do that.

File: libs/community/tests/integration_tests/vectorstores/qdrant/test_from_existing_index.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+import pytest
+
+from langchain_community.vectorstores import Qdrant
+from tests.integration_tests.vectorstores.fake_embeddings import (
+    ConsistentFakeEmbeddings,
+)
+
+
+@pytest.mark.parametrize("collection_name", ["custom-collection"])
```
Comment (Anush008) [inline libs/community/tests/integration_tests/vectorstores/qdrant/test_from_existing_index.py:11]: Can be a local variable instead.

File: libs/community/tests/integration_tests/vectorstores/qdrant/test_from_existing_index.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+def test_qdrant_from_existing_index_uses_same_collection(
+    collection_name: str,
+) -> None:
+    """Test if the Qdrant.from_existing_client reuses the some client."""
```
Comment (Anush008) [inline libs/community/tests/integration_tests/vectorstores/qdrant/test_from_existing_index.py:15]:
```suggestion
    """Test if the Qdrant.from_existing_client reuses the same collection"""
```

**Comment by spike-spiegel-21** [comment]
@Anush008 Implemented all suggested improvements.

File: libs/community/tests/integration_tests/vectorstores/qdrant/test_from_existing_collection.py [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+from langchain_community.vectorstores import Qdrant
+from tests.integration_tests.vectorstores.fake_embeddings import (
+    ConsistentFakeEmbeddings,
+)
+
+
+@pytest.mark.parametrize("vector_name", ["custom-vector"])
+def test_qdrant_from_existing_collection_uses_same_collection(vector_name: str) -> None:
+    """Test if the Qdrant.from_existing_collection reuses the same client."""
```
Comment (Anush008) [inline libs/community/tests/integration_tests/vectorstores/qdrant/test_from_existing_collection.py:14]:
```suggestion
    """Test if the Qdrant.from_existing_collection reuses the same collection."""
```

File: libs/community/langchain_community/vectorstores/qdrant.py [hunk 6]
```diff
@@ -1367,6 +1367,51 @@ def from_texts(
...
+        prefer_grpc: bool = False,
+        https: Optional[bool] = None,
+        api_key: Optional[str] = None,
+        prefix: Optional[str] = None,
+        timeout: Optional[float] = None,
+        host: Optional[str] = None,
+        **kwargs: Any,
+    ) -> Qdrant:
+        """
+        Get instance of an existing Qdrant collection stored locally.
```
Comment (Anush008) [inline libs/community/langchain_community/vectorstores/qdrant.py:1389]:
```suggestion
        Get instance of an existing Qdrant collection.
```

**Review by Anush008 — APPROVED** [review]
Thank you @spike-spiegel-21.

**Pull request merged**